metadata-db-app/src/cmr/metadata_db/services/generic_documents.clj

```diff
@@ -21,53 +21,51 @@
 ;; to database column names which are to be shared with the world.
 (def info-fields [:concept-id :native-id :provider-id :document-name :schema
                   :format :mime-type :revision-id :revision-date :created-at
-                  :user-id])
+                  :user-id :concept-type :concept-sub-type])
 
-;; These are fields that should not be inclucded in the :Metadata field because
-;; they are generated by software. Add items to this list corrisponding to
-;; database column names which should NOT be shared with the world and are not
-;; in the info-fields list above.
-(def fields-to-drop (vec (concat info-fields [:id :deleted :transaction-id])))
+;; Insert-generic-document and other CRUD actions, except Read, return nil on
+;; success, To make the code more understandable, give that success a name.
+;; Fails will be handled with exceptions.
+(def success-response nil)
 
 (defn- raw-generic->response
   "Take the raw database object returned from either SQL or In-Memory and shape
    it into the expected response by pulling out generated fields like concept-id
    and putting them in the :Info field, then putting the metadata inside the
-   :Metadata field without any of these extra fields."
+   :Metadata field."
   [raw-doc]
-  ;; TODO: Generic work: if the lower levels use :extra-fields instead of mixing, then would this code be simplified?
-  {:Info (select-keys raw-doc info-fields)
-   :Metadata (reduce #(dissoc %1 %2) raw-doc fields-to-drop)})
+  {:Info (select-keys raw-doc info-fields) :Metadata (:metadata raw-doc)})
 
 (defn insert-generic-document
   "Insert a document under the provided provider-id. Generate a concept ID for
    the new record, At this time, nothing prevents multiple copies of a record
-   from being inserted, users must know what they are doing."
+   from being inserted, users must know what they are doing. Update will not call
+   this function."
   [context params provider-id raw-native-id document]
   (let [db (mdb-util/context->db context)
         ;; Validate that the provider exists.
         _ (provider-service/get-provider-by-id context provider-id true)
         document (if (map? document) (json/generate-string document) document)
         document-as-map (json/parse-string document true)
-        ;; TODO: Generic work: Fix native id so that it is passed in correctly.
         native-id (or raw-native-id (.toString (java.util.UUID/randomUUID))) ;; can this stay?
+        inner-metadata (json/parse-string (:metadata document-as-map) true)
+        doc-name (get inner-metadata :Name native-id)
         concept-type (get common-concepts/concept-prefix->concept-type (:concept-sub-type document-as-map))
         document-add (assoc document-as-map
                             :provider-id (str provider-id)
-                            :concept-type concept-type
-                            :revision-id 1
+                            :concept-type :generic
```

Comment:
This can't be :generic anymore unless the actual concept type is generic.  This will mess up the DB calls and indexer calls and you will get lots of exceptions on the indexer.